lib/spack/spack/fetch_strategy.py

```diff
@@ -336,6 +336,13 @@ def fetch(self):
             raise FailedDownloadError(url)
 
     def _existing_url(self, url):
+        # first, try the range request, if that fails
+        # (which can happen with some server configurations)
+        # try header request
+        return (self._check_existing_url_by_range_request(url) or
+                self._check_existing_url_by_header_request(url))
```

Comment:
My impression is that the names of the functions reflect the `curl` implementation, but not the `urllib` one. Can we refactor to have a function for `curl` and one for `urllib`?

Reply:
Yes, I think this PR predates the urllib implementation, I'll take a look at this issue.

Reply:
I finally managed to take a look and it seemed most appropriate to just tack my additions onto `_existing_url`. However I couldn't find a way to solve the fastjet fetching issue with urllib, so I'm not sure how much good this is going to do.